config/externalname.go

```diff
@@ -154,6 +154,17 @@ var TerraformPluginSDKExternalNameConfigs = map[string]config.ExternalName{
 	// Event data stores can be imported using their arn
 	"aws_cloudtrail_event_data_store": config.IdentifierFromProvider,
 
+	// codeartifact
+	//
+	// CodeArtifact Domain can be imported using the CodeArtifact Domain arn
+	"aws_codeartifact_domain": config.IdentifierFromProvider,
+	// CodeArtifact Domain Permissions Policies can be imported using the CodeArtifact Domain ARN
+	"aws_codeartifact_domain_permissions_policy": config.IdentifierFromProvider,
```

Comment:
If we can, it's better to construct the external name from the `spec`. It looks like with at least the domain_permissions_policy and repository_permissions_policy that should be easy enough to do.

Reply:
@mbbush do you mean something like https://github.com/crossplane-contrib/provider-upjet-aws/blob/87d323b06a409d034515026db83c87333c34cad6/config/externalname.go#L106? 

```
        "aws_prometheus_rule_group_namespace": config.TemplatedStringAsIdentifier("name", "arn:aws:aps:{{ .setup.configuration.region }}:{{ .setup.client_metadata.account_id }}:rulegroupsnamespace/{{ .parameters.workspace_id }}/{{ .external_name }}"),
```

Reply:
@mbbush I've updated the code